### DISCUSSION TITLE
Canonical naming of compactly-generated-related properties

### DISCUSSION THREAD

**Discussion by StevenClontz** [pi-base/data#520] [Ideas]
Spinning off conversation from #518.

We have three "compactly-generated" (i.e. "k-space") properties:

- https://topology.pi-base.org/properties/P000140/
- https://topology.pi-base.org/properties/P000141/
- https://topology.pi-base.org/properties/P000142/

In the literature, "generated by compact" can mean three things: compact subspaces, maps from compact-T2 subspaces, or compact-T2 subspaces.

This pattern extends to the k-Hausdorff properties. And we also have $k_\omega$: https://topology.pi-base.org/properties/P000098/

Since in the literature, $k$-space, $k$-Hausdroff, and $k_\omega$ seem to be standard (if ambiguous), I think we should follow similar patterns for our canonical names: $k_i$-space and $k_i$-Hausdorff. We can keep $k_\omega$ and not use (I guess) $(k_1)\_\omega$, at least until someone publishes on a space for which a set is closed whenever its intersection with any continuous image of a compact Hausdroff space is closed (which I guess would be $(k_2)_\omega$).

**Comment by prabau** [comment]
$k_\omega$ seems pretty standard and not very ambiguous.  I think the only difference between variants that you and I have seen about this concept was whether to consider the space as generated from compact subspaces (as in Lawson & Madison https://link.springer.com/article/10.1007/BF02194829 if I remember correctly) or from compact Hausdorff subspaces (as in the survey article by Franklin & Smith Thomas).  I think the more general choice we made in pi-base seems reasonable.

But $k$-space is another matter.  As I explained in the lead paragraph of https://en.wikipedia.org/wiki/Compactly_generated_space, some authors use "$k$-space", some use "compactly generated", there are at least two rather common ways of choosing the corresponding families of compact subspaces that generate the topology (leading to what pi-base calls CG1 and CG2) (as CG3 seems less important, with not many nice properties)), and some authors include extra separation conditions, some don't.  There is really nothing standard here.  See in particular https://math.stackexchange.com/questions/4640637/open-sets-in-compactly-generated-spaces for a sample of incompatible terminologies from various authors.  

Sometimes some people (at least even on mo) seem not totally aware of the full range of concepts and assume one thing means another.
Ideally, I would hope that someone would publish an article somewhere mentioning this unfortunate state of non-standard terminologies and introduce in one place at least what we have called CG1, CG2, CG3 or like you propose, $k_i$-space, etc.  And then one could reference that from the literature, both in pi-base and in wikipedia.  Maybe this could be done in "Topological spaces among US" by @StevenClontz and @marswill ?

**Comment by prabau** [comment]
> We can keep $k_\omega$ and not use (I guess) $(k_1)\_\omega$, at least until someone publishes on a space for which a set is closed whenever its intersection with any continuous image of a compact Hausdroff space is closed (which I guess would be $(k_2)_\omega$).

Wouldn't the $k_\omega$ from Franklin & Smith Thomas (https://topology.nipissingu.ca/tp/reprints/v02/tp02105.pdf) be $(k_3)_\omega$ ?

**Reply by StevenClontz** [reply]
yes

**Comment by prabau** [comment]
I may have misunderstood.  As far as standardizing the names in pi-base, your proposal follow a logical pattern and seems very reasonable, even if there is no standard terminology in the literature and even if $k_i$-space does not appear (yet) in the literature.

**Comment by marswill** [comment]
> Since in the literature, k-space, k-Hausdroff, and kω seem to be standard (if ambiguous), I think we should follow similar patterns for our canonical names: ki-space and ki-Hausdorff. We can keep kω and not use (I guess) (k1)ω, at least until someone publishes on a space for which a set is closed whenever its intersection with any continuous image of a compact Hausdroff space is closed (which I guess would be (k2)ω).

~So $(k_i)\_\omega$ for $i=1,2,3$, can be defined: there exists a sequence of (compacts) (images of compact Hausdorffs) (compact Hausdorffs) whose union is $X$ and that generate the topology.  In all three cases, we lose no generality assuming they are nested.  (Note we do need to assume union is $X$, otherwise we get a weaker condition that says $X$ is the disjoint union of a $(k_i)\_\omega$ space and a discrete space, which may be wlog either uncountable or empty).~

[Edit:  The above turned out to be kind of nonsense (see discussion below).  However, defining $(k_1)\_\omega$ (resp. $(k_3)\_\omega$) to mean the space is a nested union of  compact (Hausdorff) spaces  which generate the topology, the rest of this comment should still make sense.]

For what its worth, I don't trust that Franklin Smith and Thomas article outside of Hausdorff spaces.  

They assert atop page 113 that every $k_\omega$ space (by which they mean $(k_3)\_\omega$) is Hausdorff, and then at the bottom of 114 they assert "The $k_\omega$ spaces are precisely the hemicompact $k$-spaces."

Though they never actually give their definition of $k$-space, this is not reconcile-able under any of our three definitions.  While $(k_3)\_\omega$ clearly implies $k_3$ (hence all three $k_i$'s) and implies hemicompactness (since $(k_1)\_\omega$ implies hemicompactness), even a compact $k_3$ space can fail to be Hausdorff, as indicated by $\mathbb Q^*$ https://topology.pi-base.org/spaces/?q=cg3+%2Bhemicompact+%2B%7ET2

Thinking about it a little further, $\mathbb Q^*$ is clearly not a $(k_3)\_\omega$ space:

A compact Hausdorff subspace of $\mathbb Q^*$ must intersect $\mathbb Q$ in a locally compact Hausdorff space.   Therefore, if $\\{K_n\\}$ witnessed $(k_3)\_\omega$, then the sets $K_n\cap \mathbb Q$ would be locally compact, hence not have any interior in $\mathbb Q$, so we could take a sequence $x_n \in (0,\frac{1}{n})\backslash K_n$, whose union would be a non-closed set intersecting each $K_n$ in a finite (hence closed) set.

The only thing that could reconcile this that I can think of is maybe somewhere in the literature, "hemicompact" also required the spaces in the sequence to be Hausdorff?  Not very conversant in this stuff so I don't know how plausible that is.

**Reply by marswill** [reply]
I forgot my main point, which was that I agree using $k_i$ notation, along with $k_\omega$ for $(k_1)_\omega$, seems good from a relative newcomer's perspective to all of this - it makes a lot of this clear in my mind.

**Reply by marswill** [reply]
Also, assuming $(k_3)\_\omega$ really does imply Hausdorff* then there isn't much to be gained from making it its own thing, since it's then just equivalent to $k_\omega+T_2$.

[*Edit: it does, couldn't dig up the reference but proof of this is easy, just take increasing union of closed disjoint neighborhoods separating two points in each $K_n$, using normality of each $K_n$]

**Reply by marswill** [reply]
On second thought, on math.SE I was just pointed to this [quite recent reference](https://www.cambridge.org/core/journals/canadian-mathematical-bulletin/article/pure-quotients-and-moritas-theorem-for-komega-spaces/BEFA5B925481B43EC99EA2D772A5366B) that includes $T_2$ assumption in $k_\omega$.  So maybe it should be its own thing.  

One possibility would be to include $k_\omega (\text{Hausdorff})$ and $k_\omega(\text{generalized})$, or something to that effect, to make it a little more clear to users which one is being referred to.

**Comment by prabau** [comment]
One detail about the formulation of $(k_i)\omega$ that you mentioned above:
> "there exists a sequence of (compacts) (images of compact Hausdorffs) (compact Hausdorffs) whose union is and that generate the topology.  In all three cases, we lose no generality assuming they are nested. "

For $i=1$ (i.e. the space is generated by countably many compact subspaces that cover $X$):  It is clear that we can assume wlog that the compact sets are nested, since the union of two compact sets is compact.

But for $i=3$ (i.e. the space is generated by countably many compact Haudorff subspaces that cover $X$) we cannot assume that the compact sets are nested, as the union of two compact T2 subspaces need not be T2.

For an example, take $X=[0,1]\cup\\{1^\ast\\}$ with the "telophase topology" (https://topology.pi-base.org/spaces/S000065).  The space is the union of two compact T2 sets and the topology is the final topology wrt the inclusions of these two subspaces.  But there is no ascending sequence of compact T2 subspaces that would generate the topology.  $X$ is the union of the subspaces $[0,1-1/n]\cup\\{1,1^\ast\\}$, each of them compact Hausdorff, but $1$ and $1^*$ are isolated in the final topology...  So $X$ would match the definition if we don't require an ascending sequence, but not otherwise.  And maybe Franklin et al. did not want $X$ to be $k_\omega$ in their sense, which is why they required an ascending sequence in their definition of $k_\omega$-decomposition.

As for $i=2$, it could be a little more subtle.  More below.

**Reply by marswill** [reply]
Oh yes, that's quite right.  For $i=2$ though it's fine I think, since if $A, B$ are images of compact Hausdorff spaces, then there is a map from disjoint union onto $A\cup B$, which is what I had in mind.  I must have completely forgotten about $i=3$ case when making that comment.

**Reply by marswill** [reply]
Oh maybe there is indeed another problem, I somehow talked myself into thinking its enough to look at the intersection with the images of compact Hausdorff spaces, rather than the pre-images.  But I can't seem to reporduce my argument for that so that's probably not true either.

**Reply by prabau** [reply]
Yeah, that's exactly what I was in the middle of writing about.

**Reply by prabau** [reply]
For $i=2$ I was also thinking of the map from the disjoint union of compact T2 spaces, but I am thinking it could be a little more subtle.

Back to the definition of $k_2$-space, which is a space where every $k_2$-closed set is closed.  And $A\subseteq X$ is $k_2$-closed if $u^{-1}(A)$ is closed in $K$ for every (continuous) "test map" $u:K\to X$ with $K$ compact Hausdorff space.

If I am not mistaken, that does not seem to be the same as saying $X$ is [coherent](https://en.wikipedia.org/wiki/Coherent_topology) with the collection of images of compact Hausdorff spaces.

If no immediate example comes to mind, it would be worth asking on mathse.  I think I may have seen a counterexample, but I don't remember it now.  Maybe I added it to https://en.wikipedia.org/wiki/Compactly_generated_space.  Need to check.

**Reply by prabau** [reply]
https://math.stackexchange.com/questions/4846810/about-the-definition-of-k-closed-set-cg2
